File: app/src/main/java/com/coinv/app/ui/profile/ProfileScreen.kt
package com.coinv.app.ui.profile

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.Switch
import androidx.compose.material3.SwitchDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.hilt.navigation.compose.hiltViewModel
import com.coinv.app.ui.components.SectionHeader
import com.coinv.app.ui.components.SurfaceCard
import com.coinv.app.ui.theme.CoinBackground
import com.coinv.app.ui.theme.CoinBlue
import com.coinv.app.ui.theme.CoinBorder
import com.coinv.app.ui.theme.CoinChrome
import com.coinv.app.ui.theme.CoinChromeMuted
import com.coinv.app.ui.theme.CoinError
import com.coinv.app.ui.theme.CoinWarning
import com.coinv.app.ui.theme.CoinSuccess
import com.coinv.app.ui.theme.CoinSurface
import com.coinv.app.ui.theme.jetBrainsMono

private val INTEGRATIONS = listOf(
    "Google Calendar", "Gmail", "Notion", "Slack", "GitHub", "Zoom"
)

@Composable
fun ProfileScreen(
    onStopMonitoring: () -> Unit = {},
    onOpenAboutMe: () -> Unit = {},
    modifier: Modifier = Modifier,
    viewModel: ProfileViewModel = hiltViewModel()
) {
    val state by viewModel.uiState.collectAsState()
    val profile = state.profile
    val settings = state.settings

    LazyColumn(
        modifier = modifier
            .fillMaxSize()
            .background(CoinBackground),
        contentPadding = PaddingValues(bottom = 24.dp)
    ) {
        item {
            SectionHeader("Profile")
            SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp)) {
                var editingName by remember { mutableStateOf(false) }
                var nameInput by remember { mutableStateOf(profile?.name ?: "Shyam") }
                if (editingName) {
                    OutlinedTextField(
                        value = nameInput,
                        onValueChange = { nameInput = it },
                        label = { Text("Your name") },
                        modifier = Modifier.fillMaxWidth()
                    )
                    TextButton(onClick = {
                        viewModel.updateName(nameInput)
                        editingName = false
                    }) { Text("Save", color = CoinBlue) }
                } else {
                    Text(profile?.name ?: "Shyam", style = MaterialTheme.typography.headlineMedium, color = CoinChrome)
                    TextButton(onClick = { editingName = true; nameInput = profile?.name ?: "Shyam" }) {
                        Text("Edit name", color = CoinBlue, fontSize = 12.sp)
                    }
                }
                Text(
                    "Cognitive State: ${profile?.cognitiveState ?: "Ready"}",
                    fontFamily = jetBrainsMono,
                    fontSize = 12.sp,
                    color = CoinBlue,
                    modifier = Modifier.padding(top = 4.dp)
                )
            }
        }

        item { SectionHeader("AI Coaches") }
        item {
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                AI_COACHES.take(3).forEach { coach ->
                    CoachChip(coach, profile?.activeCoach == coach) { viewModel.selectCoach(coach) }
                }
            }
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 8.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                AI_COACHES.drop(3).forEach { coach ->
                    CoachChip(coach, profile?.activeCoach == coach) { viewModel.selectCoach(coach) }
                }
            }
        }

        item { SectionHeader("Assistant Modes") }
        item {
            SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp)) {
                Text(
                    "Listening — responds when you ask. Monitoring — collects context silently.",
                    fontSize = 12.sp,
                    color = CoinChromeMuted
                )
                Text(
                    "Headset: 1× play/pause = Listening · 2× = Monitoring",
                    fontFamily = jetBrainsMono,
                    fontSize = 11.sp,
                    color = CoinBlue,
                    modifier = Modifier.padding(top = 8.dp)
                )
            }
        }

        item { SectionHeader("Privacy & Monitoring") }
        item {
            SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp)) {
                SettingRow("Allow monitoring mode") {
                    Switch(
                        checked = settings.monitoringEnabled,
                        onCheckedChange = { viewModel.setMonitoringEnabled(it) },
                        colors = SwitchDefaults.colors(checkedThumbColor = CoinBlue)
                    )
                }
                SettingRow("Local-only processing") {
                    Switch(
                        checked = settings.localOnlyProcessing,
                        onCheckedChange = { viewModel.setLocalOnlyProcessing(it) },
                        colors = SwitchDefaults.colors(checkedThumbColor = CoinBlue)
                    )
                }
                SettingRow("Stop monitoring now") {
                    TextButton(onClick = onStopMonitoring) {
                        Text("Disable", color = CoinWarning, fontSize = 12.sp)
                    }
                }
                Text(
                    "Monitoring shows a visible indicator and never speaks unless you ask directly.",
                    fontSize = 11.sp,
                    color = CoinChromeMuted,
                    modifier = Modifier.padding(top = 4.dp)
                )
            }
        }

        item { SectionHeader("Legacy Listening") }
        item {
            val modes = listOf(
                "off" to "Off",
                "push_to_talk" to "Push To Talk",
                "wake_word" to "Wake Word",
                "always_listening" to "Always On"
            )
            Column(
                modifier = Modifier.padding(horizontal = 20.dp),
                verticalArrangement = Arrangement.spacedBy(6.dp)
            ) {
                modes.chunked(2).forEach { row ->
                    Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                        row.forEach { (mode, label) ->
                            FilterChip(
                                selected = profile?.listeningMode == mode,
                                onClick = { viewModel.updateListeningMode(mode) },
                                label = { Text(label, fontSize = 11.sp) },
                                colors = FilterChipDefaults.filterChipColors(
                                    selectedContainerColor = CoinSurface,
                                    selectedLabelColor = CoinBlue,
                                    containerColor = CoinBackground,
                                    labelColor = CoinChromeMuted
                                ),
                                border = FilterChipDefaults.filterChipBorder(
                                    enabled = true,
                                    selected = profile?.listeningMode == mode,
                                    borderColor = CoinBorder,
                                    selectedBorderColor = CoinBlue
                                )
                            )
                        }
                    }
                }
                Text(
                    "A microphone indicator appears when background listening is active.",
                    fontSize = 11.sp,
                    color = CoinChromeMuted
                )
            }
        }

        if (state.learning.isNotEmpty()) {
            item { SectionHeader("Learning Hub") }
            items(state.learning, key = { it.id }) { item ->
                SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp)) {
                    Text(item.topic, style = MaterialTheme.typography.titleMedium, color = CoinChrome)
                    Text(item.summary, style = MaterialTheme.typography.bodySmall, color = CoinChromeMuted)
                    Text(
                        "${item.progress}% · ${item.flashcardsCount} flashcards",
                        fontFamily = jetBrainsMono,
                        fontSize = 11.sp,
                        color = CoinSuccess,
                        modifier = Modifier.padding(top = 4.dp)
                    )
                }
            }
        }

        item { SectionHeader("Integrations") }
        item {
            Text(
                "Coming soon — connect your tools to enrich cognitive context.",
                color = CoinChromeMuted,
                fontSize = 12.sp,
                modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp)
            )
            Row(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(horizontal = 20.dp, vertical = 4.dp),
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                INTEGRATIONS.take(3).forEach { name ->
                    Text(name, fontSize = 11.sp, color = CoinChromeMuted)
                }
            }
        }

        item { SectionHeader("About Me") }
        item {
            SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp)) {
                Text(
                    "Tell CoinV about yourself — stored as value memories and injected into every AI call.",
                    fontSize = 12.sp,
                    color = CoinChromeMuted
                )
                TextButton(onClick = onOpenAboutMe) {
                    Text("Edit About Me", color = CoinBlue, fontSize = 12.sp)
                }
            }
        }

        item { SectionHeader("What CoinV knows about you") }
        item {
            val valueMemories by viewModel.valueMemories.collectAsState()
            SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp)) {
                if (valueMemories.isEmpty()) {
                    Text(
                        "No value memories yet. Fill in About Me to verify wiring.",
                        fontSize = 12.sp,
                        color = CoinChromeMuted
                    )
                } else {
                    valueMemories.forEach { memory ->
                        Text(
                            memory.content,
                            style = MaterialTheme.typography.bodySmall,
                            color = CoinChrome,
                            modifier = Modifier.padding(vertical = 4.dp)
                        )
                    }
                }
            }
        }

        item { SectionHeader("Active behaviors") }
        item {
            val behaviorStats by viewModel.behaviorStats.collectAsState()
            SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp)) {
                if (behaviorStats.isEmpty()) {
                    Text("Loading behavior stats…", fontSize = 12.sp, color = CoinChromeMuted)
                } else {
                    behaviorStats.forEach { stat ->
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(vertical = 6.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column(modifier = Modifier.weight(1f)) {
                                Text(stat.label, color = CoinChrome, style = MaterialTheme.typography.bodyMedium)
                                Text(
                                    "${stat.shownCount} shown · ${stat.dismissedCount} dismissed (30d)",
                                    fontSize = 11.sp,
                                    color = CoinChromeMuted,
                                    fontFamily = jetBrainsMono
                                )
                            }
                            Text(
                                text = if (stat.isActive) "active" else "quiet for you",
                                fontSize = 10.sp,
                                fontFamily = jetBrainsMono,
                                color = if (stat.isActive) CoinBlue else CoinChromeMuted
                            )
                        }
                    }
                }
                TextButton(onClick = { viewModel.refreshBehaviorStats() }) {
                    Text("Refresh", color = CoinBlue, fontSize = 12.sp)
                }
            }
        }

        item { SectionHeader("Settings") }
        item {
            Text(
                text = "Memories: ${state.semanticMemoryCount} stored",
                color = CoinBlue,
                fontSize = 12.sp,
                fontFamily = jetBrainsMono,
                modifier = Modifier.padding(horizontal = 20.dp, vertical = 4.dp)
            )
        }
        item {
            SurfaceCard(modifier = Modifier.padding(horizontal = 20.dp)) {
                SettingRow("Theme") {
                    Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                        listOf("dark" to "Dark", "light" to "Light").forEach { (value, label) ->
                            FilterChip(
                                selected = settings.theme == value,
                                onClick = { viewModel.setTheme(value) },
                                label = { Text(label, fontSize = 11.sp) }
                            )
                        }
                    }
                }
                SettingRow("Privacy analytics") {
                    Switch(
                        checked = settings.privacyAnalytics,
                        onCheckedChange = { viewModel.setPrivacyAnalytics(it) },
                        colors = SwitchDefaults.colors(checkedThumbColor = CoinBlue)
                    )
                }
                SettingRow("Notifications") {
                    Switch(
                        checked = settings.notificationsEnabled,
                        onCheckedChange = { viewModel.setNotifications(it) },
                        colors = SwitchDefaults.colors(checkedThumbColor = CoinBlue)
                    )
                }
                SettingRow("Data export") {
                    TextButton(onClick = { viewModel.exportData() }) {
                        Text("Export profile", color = CoinBlue, fontSize = 12.sp)
                    }
                }
                SettingRow("Clear memory") {
                    TextButton(onClick = { viewModel.clearMemory() }) {
                        Text("Clear all", color = CoinWarning, fontSize = 12.sp)
                    }
                }
                SettingRow("Reset app") {
                    TextButton(onClick = { viewModel.resetApp() }) {
                        Text("Reset", color = CoinError, fontSize = 12.sp)
                    }
                }
            }
        }
    }
}

@Composable
private fun SettingRow(label: String, content: @Composable () -> Unit) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = 8.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        Text(label, style = MaterialTheme.typography.bodyMedium, color = CoinChrome)
        content()
    }
}

@Composable
private fun CoachChip(name: String, selected: Boolean, onClick: () -> Unit) {
    FilterChip(
        selected = selected,
        onClick = onClick,
        label = { Text(name, fontSize = 11.sp) },
        colors = FilterChipDefaults.filterChipColors(
            selectedContainerColor = CoinSurface,
            selectedLabelColor = CoinBlue,
            containerColor = CoinBackground,
            labelColor = CoinChromeMuted
        ),
        border = FilterChipDefaults.filterChipBorder(
            enabled = true,
            selected = selected,
            borderColor = CoinBorder,
            selectedBorderColor = CoinBlue
        )
    )
}

File: lib/ui/profile_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../core/coinv_theme.dart';
import '../state/coinv_controller.dart';
import 'coin_widgets.dart';

const _coaches = [
  'Founder Coach',
  'Productivity Coach',
  'Learning Coach',
  'Career Coach',
  'Thinking Coach',
  'Decision Coach',
];

class ProfileScreen extends StatelessWidget {
  const ProfileScreen({required this.onAboutMe, super.key});
  final VoidCallback onAboutMe;

  @override
  Widget build(BuildContext context) {
    final state = context.watch<CoinVController>();
    final settings = state.settings;
    final name = state.userProfile['name'] as String? ?? 'You';
    return ListView(
      padding: const EdgeInsets.only(bottom: 30),
      children: [
        const SectionTitle('Profile'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: CoinCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                Text(name, style: Theme.of(context).textTheme.headlineSmall),
                TextButton(
                  onPressed: () => _editName(context, name),
                  child: const Text('Edit name'),
                ),
                Text(
                  'Cognitive State: '
                  '${state.userProfile['cognitive_state'] ?? 'Ready'}',
                  style: const TextStyle(
                    color: CoinColors.blue,
                    fontFamily: 'monospace',
                  ),
                ),
              ],
            ),
          ),
        ),
        const SectionTitle('AI Coaches'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: Wrap(
            spacing: 8,
            runSpacing: 6,
            children: _coaches
                .map(
                  (coach) => ChoiceChip(
                    label: Text(coach),
                    selected: state.userProfile['active_coach'] == coach,
                    onSelected: (_) => state.setCoach(coach),
                  ),
                )
                .toList(),
          ),
        ),
        const SectionTitle('Privacy & Monitoring'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: CoinCard(
            child: Column(
              children: [
                _SwitchRow(
                  label: 'Allow monitoring mode',
                  value: settings.monitoringEnabled,
                  onChanged: (value) => state.updateSettings(
                    settings.copyWith(monitoringEnabled: value),
                  ),
                ),
                _SwitchRow(
                  label: 'Local-only processing',
                  value: settings.localOnlyProcessing,
                  onChanged: (value) => state.updateSettings(
                    settings.copyWith(localOnlyProcessing: value),
                  ),
                ),
                ListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Stop monitoring now'),
                  trailing: TextButton(
                    onPressed: () => state.enterIdle('profile'),
                    child: const Text(
                      'Disable',
                      style: TextStyle(color: CoinColors.warning),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ),
        const SectionTitle('Listening'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: Wrap(
            spacing: 8,
            children:
                const [
                  ('off', 'Off'),
                  ('push_to_talk', 'Push To Talk'),
                  ('wake_word', 'Wake Word'),
                  ('always_listening', 'Always On'),
                ].map((item) {
                  final selected =
                      state.userProfile['listening_mode'] == item.$1;
                  return ChoiceChip(
                    label: Text(item.$2),
                    selected: selected,
                    onSelected: (_) => state.updateListeningMode(item.$1),
                  );
                }).toList(),
          ),
        ),
        const SectionTitle('About Me'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: CoinCard(
            child: Column(
              crossAxisAlignment: CrossAxisAlignment.start,
              children: [
                const Text(
                  'Saved as value memories and injected into every AI call.',
                ),
                TextButton(
                  onPressed: onAboutMe,
                  child: const Text('Edit About Me'),
                ),
              ],
            ),
          ),
        ),
        const SectionTitle('What CoinV knows about you'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: CoinCard(
            child: state.valueMemories.isEmpty
                ? const Text('No value memories yet.')
                : Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: state.valueMemories
                        .map(
                          (memory) => Padding(
                            padding: const EdgeInsets.symmetric(vertical: 4),
                            child: Text(memory.content),
                          ),
                        )
                        .toList(),
                  ),
          ),
        ),
        const SectionTitle('Active Behaviors'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: CoinCard(
            child: Column(
              children: state.behaviorStats
                  .map(
                    (stat) => ListTile(
                      contentPadding: EdgeInsets.zero,
                      title: Text(_behaviorLabel(stat.type)),
                      subtitle: Text(
                        '${stat.shown} shown · ${stat.dismissed} dismissed',
                      ),
                      trailing: Text(
                        stat.active ? 'active' : 'quiet for you',
                        style: TextStyle(
                          color: stat.active
                              ? CoinColors.blue
                              : CoinColors.muted,
                          fontFamily: 'monospace',
                          fontSize: 11,
                        ),
                      ),
                    ),
                  )
                  .toList(),
            ),
          ),
        ),
        const SectionTitle('Settings'),
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 20),
          child: CoinCard(
            child: Column(
              children: [
                ListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Theme'),
                  trailing: SegmentedButton<String>(
                    segments: const [
                      ButtonSegment(value: 'dark', label: Text('Dark')),
                      ButtonSegment(value: 'light', label: Text('Light')),
                      ButtonSegment(value: 'system', label: Text('System')),
                    ],
                    selected: {settings.theme},
                    onSelectionChanged: (selection) => state.updateSettings(
                      settings.copyWith(theme: selection.first),
                    ),
                  ),
                ),
                _SwitchRow(
                  label: 'Privacy analytics',
                  value: settings.privacyAnalytics,
                  onChanged: (value) => state.updateSettings(
                    settings.copyWith(privacyAnalytics: value),
                  ),
                ),
                _SwitchRow(
                  label: 'Notifications',
                  value: settings.notificationsEnabled,
                  onChanged: (value) => state.updateSettings(
                    settings.copyWith(notificationsEnabled: value),
                  ),
                ),
                ListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Memory retention'),
                  subtitle: const Text('Value memories are always retained'),
                  trailing: DropdownButton<int>(
                    value: settings.memoryRetentionDays,
                    items: const [
                      DropdownMenuItem(value: 30, child: Text('30 days')),
                      DropdownMenuItem(value: 90, child: Text('90 days')),
                      DropdownMenuItem(value: 365, child: Text('1 year')),
                      DropdownMenuItem(value: 3650, child: Text('Keep')),
                    ],
                    onChanged: (value) {
                      if (value != null) {
                        state.updateSettings(
                          settings.copyWith(memoryRetentionDays: value),
                        );
                      }
                    },
                  ),
                ),
                ListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Data export'),
                  trailing: TextButton(
                    onPressed: state.exportData,
                    child: const Text('Export'),
                  ),
                ),
                ListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Clear memory'),
                  trailing: TextButton(
                    onPressed: () => _confirm(
                      context,
                      'Clear all memories and About Me facts?',
                      state.clearMemory,
                    ),
                    child: const Text(
                      'Clear all',
                      style: TextStyle(color: CoinColors.warning),
                    ),
                  ),
                ),
                ListTile(
                  contentPadding: EdgeInsets.zero,
                  title: const Text('Reset app'),
                  trailing: TextButton(
                    onPressed: () => _confirm(
                      context,
                      'Reset all CoinV data and settings?',
                      state.resetApp,
                    ),
                    child: const Text(
                      'Reset',
                      style: TextStyle(color: CoinColors.error),
                    ),
                  ),
                ),
              ],
            ),
          ),
        ),
      ],
    );
  }

  Future<void> _editName(BuildContext context, String existing) async {
    final controller = TextEditingController(text: existing);
    final state = context.read<CoinVController>();
    await showDialog<void>(
      context: context,
      builder: (dialogContext) => AlertDialog(
        title: const Text('Your name'),
        content: TextField(controller: controller, autofocus: true),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(dialogContext),
            child: const Text('Cancel'),
          ),
          FilledButton(
            onPressed: () async {
              await state.updateName(controller.text);
              if (dialogContext.mounted) Navigator.pop(dialogContext);
            },
            child: const Text('Save'),
          ),
        ],
      ),
    );
    controller.dispose();
  }

  Future<void> _confirm(
    BuildContext context,
    String prompt,
    Future<void> Function() action,
  ) async {
    final confirmed = await showDialog<bool>(
      context: context,
      builder: (dialogContext) => AlertDialog(
        title: Text(prompt),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(dialogContext, false),
            child: const Text('Cancel'),
          ),
          FilledButton(
            onPressed: () => Navigator.pop(dialogContext, true),
            child: const Text('Confirm'),
          ),
        ],
      ),
    );
    if (confirmed == true) await action();
  }
}

class _SwitchRow extends StatelessWidget {
  const _SwitchRow({
    required this.label,
    required this.value,
    required this.onChanged,
  });
  final String label;
  final bool value;
  final ValueChanged<bool> onChanged;

  @override
  Widget build(BuildContext context) => SwitchListTile(
    contentPadding: EdgeInsets.zero,
    title: Text(label),
    value: value,
    onChanged: onChanged,
  );
}

String _behaviorLabel(String type) => switch (type) {
  'devils_advocate' => "Devil's advocate",
  'bias_spotter' => 'Bias spotter',
  'promise_tracker' => 'Promise tracker',
  'commitment_guard' => 'Commitment guard',
  'decision_followup' => 'Decision follow-up',
  _ => type,
};
